### PR TITLE
Deprecated Series.ftype, Series.ftypes and DataFrame.ftypes features

### LINKED ISSUE #26705: Deprecate Series.ftype / DataFrame.ftypes
With the `ftype(s)` property, you could check if data was sparse:

```
In [26]: df = pd.DataFrame({'a': pd.SparseArray([1, np.nan, 1])})

In [27]: df
Out[27]: 
     a
0  1.0
1  NaN
2  1.0

In [28]: df.dtypes
Out[28]: 
a    Sparse[float64, nan]
dtype: object

In [29]: df.ftypes
Out[29]: 
a    float64:sparse
dtype: object
```

But now that the dtype itself is a EA SparseDtype, I don't think those `ftype` and `ftypes` attributes serve any good purpose. So let's deprecate them.

### PR DESCRIPTION
**Pull request by thiviyanT**
Deprecation of `Series.ftype`, `Series.ftypes` and `DataFrame.ftypes` 
closes #26705

- [x] Updated the docstrings for the sphinx documentation
- [x] Updated the unit tests for testing deprecation warnings

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -419,14 +419,30 @@ def dtypes(self):
     def ftype(self):
         """
         Return if the data is sparse|dense.
+
+        .. deprecated:: 0.25.0
+           Use :func:`dtype` instead.
```
Comment (jreback) [inline pandas/core/series.py]: @datapythonista  will these render?

File: pandas/tests/series/test_dtypes.py [hunk 2]
```diff
@@ -48,8 +48,10 @@ def test_dtype(self, datetime_series):
 
         assert datetime_series.dtype == np.dtype('float64')
         assert datetime_series.dtypes == np.dtype('float64')
-        assert datetime_series.ftype == 'float64:dense'
-        assert datetime_series.ftypes == 'float64:dense'
+        with tm.assert_produces_warning(FutureWarning):
```
Comment (jreback) [inline pandas/tests/series/test_dtypes.py]: can you add the comment with this issue number (and a blank line before)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 1 above)
Comment (datapythonista) [reply pandas/core/series.py]: yes, I think this is correct and should render as a box like the one in http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_table.html

File: pandas/tests/series/test_dtypes.py (hunk 2 above)
Comment (ThiviyanThanapalasingam) [reply pandas/tests/series/test_dtypes.py]: Sure.

**Comment by ThiviyanThanapalasingam** [comment]
@jreback @jorisvandenbossche - Any idea why these builds are failing?

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, pls add a note in the whatsnew Deprecation section.

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -478,6 +478,8 @@ Other Deprecations
 - The :meth:`DataFrame.compound` and :meth:`Series.compound` methods are deprecated and will be removed in a future version (:issue:`26405`).
 - The internal attributes ``_start``, ``_stop`` and ``_step`` attributes of :class:`RangeIndex` have been deprecated.
   Use the public attributes :attr:`~RangeIndex.start`, :attr:`~RangeIndex.stop` and :attr:`~RangeIndex.step` instead (:issue:`26581`).
+- The :meth:`Series.ftype`, :meth:`Series.ftypes` and :meth:`DataFrame.ftypes` methods are deprecated and will be removed in a future version.
+  Instead, use :meth:`Series.dtype`, :meth:`Series.dtypes` and :meth:`DataFrame.dtypes` (:issue:`26705`).
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.25.0.rst]:
```suggestion
  Instead, use :meth:`Series.dtype` and :meth:`DataFrame.dtypes` (:issue:`26705`).
```

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5562,6 +5565,11 @@ def ftypes(self):
         3    float64:sparse
         dtype: object
         """
+        warnings.warn("ftypes is deprecated and will "
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]:
```suggestion
        warnings.warn("DataFrame.ftypes is deprecated and will "
```

File: pandas/core/generic.py [hunk 5]
```diff
@@ -5562,6 +5565,11 @@ def ftypes(self):
...
+                      "be removed in a future version. "
+                      "Use dtypes instead.",
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]:
```suggestion
                      "Use DataFrame.dtypes instead.",
```

File: pandas/core/series.py [hunk 6]
```diff
@@ -419,14 +419,30 @@ def dtypes(self):
...
         """
+        warnings.warn("ftype is deprecated and will "
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]:
```suggestion
        warnings.warn("Series.ftype is deprecated and will "
```

File: pandas/core/series.py [hunk 7]
```diff
@@ -419,14 +419,30 @@ def dtypes(self):
...
+                      "be removed in a future version. "
+                      "Use dtype instead.",
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]:
```suggestion
                      "Use Series.dtype instead.",
```

File: pandas/core/series.py [hunk 8]
```diff
@@ -419,14 +419,30 @@ def dtypes(self):
...
 
     @property
     def ftypes(self):
         """
         Return if the data is sparse|dense.
+
+        .. deprecated:: 0.25.0
+           Use :func:`dtypes` instead.
         """
+        warnings.warn("ftypes is deprecated and will "
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]:
```suggestion
        warnings.warn("Series.ftypes is deprecated and will "
```

File: pandas/core/series.py [hunk 9]
```diff
@@ -419,14 +419,30 @@ def dtypes(self):
...
+                      "be removed in a future version. "
+                      "Use dtypes instead.",
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]:
```suggestion
                      "Use Series.dtype instead.",
```

(dtype and dtypes are equivalent for Series, but I think we should point people to use dtype)

**Review by jorisvandenbossche — COMMENTED** [review]
@ThiviyanThanapalasingam thanks for the PR! 
I added a bunch of small suggestions (just some wording to be more explicit), normally you should be able to accept those suggestions through the github UI. 

Looks good for the rest!

**Comment by ThiviyanThanapalasingam** [comment]
Great! Appreciate the feedback.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ThiviyanThanapalasingam

**Pull request merged**